ui/media/js/dnd.js

```diff
@@ -0,0 +1,385 @@
+                val = str.slice(name.length + 1)
+                str = ""
+            }
+        } else {
+            // User formatted, use regex to get all cases, but slower.
+            const reName = new RegExp(`${name}\\s*:\\s*(.*)(?:\\r\\n|\\r|\\n)*`, 'igm')
+            const match = reName.exec(str);
+            if (match) {
+                str = str.slice(0, match.index) + str.slice(match.index + match[0].length)
+                val = match[1]
+            }
+        }
+        if (val) {
+            taskReqBody[key] = TASK_MAPPING[key].parse(val.trim())
+            console.log(TASK_MAPPING[key].name + ':', taskReqBody[key])
+            if (!str) {
+                break
+            }
+        }
+    }
+    if (Object.keys(taskReqBody).length <= 0) {
+        return undefined
+    }
+    const task = { reqBody: taskReqBody }
+    if ('seed' in taskReqBody) {
+        task.seed = taskReqBody.seed
+    }
+    return task
+}
+
+async function readFile(file, i) {
+    const fileContent = (await file.text()).trim()
+
+    // JSON File.
+    if (fileContent.startsWith('{') && fileContent.endsWith('}')) {
+        try {
+            const task = JSON.parse(fileContent)
+            restoreTaskToUI(task)
+        } catch (e) {
+            console.warn(`file[${i}]:${file.name} - File couldn't be parsed.`, e)
+        }
+        return
+    }
+
+    // Normal txt file.
+    const task = parseTaskFromText(fileContent)
+    if (task) {
+        restoreTaskToUI(task)
+    } else {
+        console.warn(`file[${i}]:${file.name} - File couldn't be parsed.`)
+    }
+}
+
+function dropHandler(ev) {
+    console.log('Content dropped...')
+    // Prevent default behavior (Prevent file/content from being opened)
+    ev.preventDefault()
+
+    if (ev?.dataTransfer?.items) { // Use DataTransferItemList interface
+        Array.from(ev.dataTransfer.items).forEach(function(item, i) {
+            if (item.kind === 'file') {
+                const file = item.getAsFile()
+                readFile(file, i)
+            }
+        })
+    } else if (ev?.dataTransfer?.files) { // Use DataTransfer interface
+        Array.from(ev.dataTransfer.files).forEach(readFile)
+    }
+}
+function dragOverHandler(ev) {
+    console.log('Content in drop zone')
+
+    // Prevent default behavior (Prevent file/content from being opened)
+    ev.preventDefault()
+
+    ev.dataTransfer.dropEffect = "copy"
+
+    let img = new Image()
+    img.src = location.host + '/media/images/favicon-32x32.png'
+    ev.dataTransfer.setDragImage(img, 16, 16)
+}
+
+document.addEventListener("drop", dropHandler)
+document.addEventListener("dragover", dragOverHandler)
+
+// Adds a copy icon if the browser grants permission to write to clipboard.
+function checkWriteToClipboardPermission (result) {
+    if (result.state == "granted" || result.state == "prompt") {
+        const style = document.createElement('style');
+        style.textContent = `
+        #copy-image-settings {
```

Comment:
Probably better to keep this in the CSS anyway. If we ever need to style this, it'll be unintuitive for the developer to look in the JS file to find where the 'copy' button is styled.